src/coreclr/tools/aot/Mono.Linker.Tests/TestCasesRunner/AssemblyChecker.cs

```diff
@@ -20,17 +23,65 @@ namespace Mono.Linker.Tests.TestCasesRunner
+
+			public AssemblyQualifiedToken (TypeSystemEntity entity) =>
+				(AssemblyName, Token) = entity switch {
+					EcmaType type => (type.Module.Assembly.GetName ().Name, MetadataTokens.GetToken (type.Handle)),
+					EcmaMethod method => (method.Module.Assembly.GetName ().Name, MetadataTokens.GetToken (method.Handle)),
+					PropertyPseudoDesc property => (((EcmaType) property.OwningType).Module.Assembly.GetName ().Name, MetadataTokens.GetToken (property.Handle)),
+					EventPseudoDesc @event => (((EcmaType) @event.OwningType).Module.Assembly.GetName ().Name, MetadataTokens.GetToken (@event.Handle)),
+					_ => (null, 0)
+				};
+
+			public AssemblyQualifiedToken (IMemberDefinition member) =>
+				(AssemblyName, Token) = member switch {
+					TypeDefinition type => (type.Module.Assembly.Name.Name, type.MetadataToken.ToInt32 ()),
+					MethodDefinition method => (method.Module.Assembly.Name.Name, method.MetadataToken.ToInt32 ()),
+					PropertyDefinition property => (property.Module.Assembly.Name.Name, property.MetadataToken.ToInt32 ()),
+					EventDefinition @event => (@event.Module.Assembly.Name.Name, @event.MetadataToken.ToInt32 ()),
+					_ => (null, 0)
+				};
+
+			public override int GetHashCode () => AssemblyName == null ? 0 : AssemblyName.GetHashCode () ^ Token.GetHashCode ();
+			public override string ToString () => $"{AssemblyName}: {Token}";
+			public bool Equals (AssemblyQualifiedToken other) =>
+				string.CompareOrdinal (AssemblyName, other.AssemblyName) == 0 && Token == other.Token;
+			public override bool Equals ([NotNullWhen (true)] object? obj) => ((AssemblyQualifiedToken?) obj)?.Equals (this) == true;
+
+			public bool IsNil => AssemblyName == null;
+		}
+
+		private static AssemblyQualifiedToken GetToken (TypeDefinition typeDefinition)
+			=> new AssemblyQualifiedToken (typeDefinition.Module.Assembly.Name.Name, typeDefinition.MetadataToken.ToInt32 ());
+
 		private readonly BaseAssemblyResolver originalsResolver;
 		private readonly ReaderParameters originalReaderParameters;
 		private readonly AssemblyDefinition originalAssembly;
 		private readonly ILCompilerTestCaseResult testResult;
 
-		private readonly Dictionary<string, TypeSystemEntity> linkedMembers;
+		private readonly Dictionary<AssemblyQualifiedToken, TypeSystemEntity> linkedMembers;
 		private readonly HashSet<string> verifiedGeneratedFields = new HashSet<string> ();
 		private readonly HashSet<string> verifiedEventMethods = new HashSet<string> ();
 		private readonly HashSet<string> verifiedGeneratedTypes = new HashSet<string> ();
 		private bool checkNames;
 
+		// Note: It's enough to exclude the type name, all of its members will also be excluded then
+		private static readonly HashSet<string> ExcludeDisplayNames = new () {
+				// Ignore compiler injected attributes to describe language version
+				"Microsoft.CodeAnalysis.EmbeddedAttribute",
+				"System.Runtime.CompilerServices.RefSafetyRulesAttribute",
```

Comment:
I agree - but I think it's worth listing it here still.
I actually think that eventually we will need to teak the current "exclude anything with System" because we have couple of tests which actually declare types in `System` namespace.
The other reason is "Documentation" - this type is "Special" in that it's declared in the test assembly, not in the framework.